modyn/selector/internal/selector_strategies/abstract_selection_strategy.py

```diff
@@ -48,12 +48,12 @@ def __init__(
         self.reset_after_trigger: bool = config["reset_after_trigger"]
 
         # weighted optimization (with weights supplied by the selector) is quite unusual, so the default us false
-        if "uses_weights" in config:
```

Comment:
These updates are needed, also the pydantic model dict dumps contain those fields with None values e.g. `{ "uses_weights": None }`. The current checks with the `in` keyword cannot handle this.